module/mempool/herocache/backdata/heropool/linkedlist_test.go

```diff
@@ -0,0 +1,88 @@
+package heropool
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestRemovingElementsFromList - first insures that list has been initialized correclty,
+// then removes elements and tests lists consistency.
+func TestRemovingElementsFromList(t *testing.T) {
+	limit_capacity := uint32(4)
+	number_of_entities := uint32(4)
+	t.Run(fmt.Sprintf("%d-limit-%d-entities", limit_capacity, number_of_entities), func(t *testing.T) {
+		withTestScenario(t, limit_capacity, number_of_entities, LRUEjection, []func(*testing.T, *Pool, []*unittest.MockEntity){
+			func(t *testing.T, pool *Pool, entities []*unittest.MockEntity) {
+				testRemovingElementsFromList(t, pool, entities, LRUEjection)
+			},
+		}...,
+		)
+	})
+}
+
+func checkList(t *testing.T, expectedList []EIndex, pool *Pool, list *state) {
+	node_index := list.head
+	i := 0
+	// check size
+	require.Equal(t, uint32(len(expectedList)), list.size)
+	// check links
+	for ; node_index != pool.free.tail; i++ {
```

Comment:
The code uses` i` as an index within a loop, but it is defined outside of the loop. It would be more readable and less error-prone if `i` was scoped within the loop: `for i := 0; node_index != pool.free.tail; i++`.

Reply:
I agree, thank you.

Reply:
As I removed this file and code in it, I propose to resolve this conversation.